doc/user/content/sql/functions/array_agg.md

```diff
@@ -0,0 +1,55 @@
+---
+title: "array_agg Function"
+description: "Aggregate values (including nulls) as an array."
+menu:
+  main:
+    parent: 'sql-functions'
+---
+
+The `array_agg(value1, value2, ...)` function aggregates values (including nulls) as an array.
+
+## Syntax
+
+{{< diagram "array-agg.svg" >}}
+
+## Signatures
+
+Parameter | Type | Description
+----------|------|------------
+_values_ | [any](../../types) | The values you want aggregated.
+
+### Return value
+
+`array_agg` returns the aggregated values as an array.
+
+## Details
+
+### Usage in dataflows
+
+While `array_agg` is available in Materialize, materializing `array_agg(values)`
+is considered an incremental view maintenance anti-pattern. Any change to the data
+underlying the function call will require the function to be recomputed entirely,
+discarding the benefits of maintaining incremental updates.
+
+Instead, we recommend that you materialize all components required for the
+`array_agg` function call and create a non-materialized view using `array_agg`
+on top of that. That pattern is illustrated in the following statements:
+
+```sql
+CREATE MATERIALIZED VIEW foo_view AS SELECT * FROM foo;
+CREATE VIEW bar AS array_agg(foo_view.bar);
```

Comment:
This doesn't work, and I'm not sure why we have it in our jsonb_agg page since it's just obviously wrong. Here's one that does work:

```
CREATE MATERIALIZED VIEW foo_view AS SELECT * FROM foo;
CREATE VIEW bar AS SELECT array_agg(foo_view.bar) FROM foo_view;
```

Reply:
Yes, it's wrong in all of them.